src/odemis/acq/stream/_live.py

```diff
@@ -365,17 +373,40 @@ def _onActive(self, active):
                 logging.exception("Failed to enable the blanker")
 
     def _startAcquisition(self, future=None):
+        # If multiple emitters are connected to a detector switch to the correct one (e.g. both a FIB and a SEM)
+        if hasVA(self._detector, "scanner") and self._detector.scanner.value != self._emitter.name:
+            self._detector.scanner.value = self._emitter.name
+
         # update Hw settings to our own ROI
         self._applyROI()
 
-        super(SEMStream, self)._startAcquisition()
+        super()._startAcquisition()
 
     def _onDwellTime(self, value):
         self._updateAcquisitionTime()
 
     def _onResolution(self, value):
         self._updateAcquisitionTime()
 
+class SEMStream(ScannerStream):
+    """ Stream containing images obtained via Scanning electron microscope.
+
+    It basically knows how to activate the scanning electron and the detector.
+    Warning: do not use local .resolution and .translation, but use the ROI.
+    Local VA .resolution is supported, but only as read-only.
+    """
+    def __init__(self, name, detector, dataflow, emitter, **kwargs):
+        if "acq_type" not in kwargs:
+            kwargs["acq_type"] = model.MD_AT_EM
+        super().__init__(name, detector, dataflow, emitter, **kwargs)
+
+
+class FIBStream(ScannerStream):
+    def __init__(self, name, detector, dataflow, emitter, **kwargs):
+        if "acq_type" not in kwargs:
+            kwargs["acq_type"] = model.MD_AT_FIB
+        super().__init__(name, detector, dataflow, emitter, **kwargs)
+        self.scale = (1.0, 1.0)  # Explicitly define the scale which isn't present as a VA on the emitter
```

Comment:
We should be able to remove this line, now that the missing scale attribute is handled.

Reply:
Isn't it nice to still define it even if it isn't essential? So it is clear for whomever uses the FIB class where scale comes from (and that it isn't actually defined) and it also prevents quite some debug messages.
What do you think?